plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoPageSource.java

```diff
@@ -191,7 +191,13 @@ else if (type.equals(REAL)) {
                     type.writeLong(output, floatToIntBits(((float) ((Number) value).doubleValue())));
                 }
                 else if (type instanceof DecimalType) {
-                    type.writeLong(output, encodeShortScaledValue(((Decimal128) value).bigDecimalValue(), ((DecimalType) type).getScale()));
+                    Decimal128 decimal = (Decimal128) value;
+                    if (decimal.compareTo(Decimal128.NEGATIVE_ZERO) == 0) {
+                        type.writeLong(output, encodeShortScaledValue(BigDecimal.ZERO, ((DecimalType) type).getScale()));
```

Comment:
Does Trino compare `DECIMAL 0.0` same as `DECIMAL -0.0`?

Reply:
Yes, it returns true. 
```sql
trino:tpch> SELECT decimal '0' = decimal '-0';
 _col0
-------
 true
(1 row)

trino:tpch> SELECT decimal '0.0' = decimal '-0.000';
 _col0
-------
 true
```